### DISCUSSION TITLE
Menyimpan & menampilkan 3 chart di 3 variabel berbeda (Capstone Project - Webscraping

### DISCUSSION THREAD

**Discussion by rannopc** [teamalgoritma/community#759] [Q&A] [answered]
Saya telah berhasil scraping data, telah di-cleansing, dan disimpan pada variabel "data_union."
Lalu saya ingin menampilkan 3 chart untuk menampilkan value tertinggi dari masing-masing kolom (imdb_rating, metascore, dan votes).
App.py:
<img width="986" alt="Screenshot 2023-09-03 at 14 10 47" src="https://github.com/teamalgoritma/community/assets/74178525/692ffeec-9437-4bc6-b4e4-d19c32d8ac69">

Index.html:
<img width="748" alt="Screenshot 2023-09-03 at 14 18 05" src="https://github.com/teamalgoritma/community/assets/74178525/ec929630-6d82-4e13-8a65-b24be44c5a49">

Namun ketika ditampilkan dalam index.html, sepertinya value yang tersimpan pada "plot_result1" "plot_result2" dan "plot_result3" sama semua, dengan bukti terlampir.

<img width="1662" alt="Screenshot 2023-09-03 at 14 11 08" src="https://github.com/teamalgoritma/community/assets/74178525/dbd09c4a-92ed-4a9a-a29b-ea47642c409b">
<img width="1662" alt="Screenshot 2023-09-03 at 14 11 15" src="https://github.com/teamalgoritma/community/assets/74178525/37420709-aff7-4dd9-a48c-527eaa4b382d">
<img width="1662" alt="Screenshot 2023-09-03 at 14 16 24" src="https://github.com/teamalgoritma/community/assets/74178525/95a1d883-415c-4370-b6ea-44146133da18">

Bagaimana cara meng-assign hasil 3 plot tadi ke variabel "plot_result1" "plot_result2" dan "plot_result3" ?
Thank you in advance.

**Comment by chchlts** [comment, marked as answer]
Dear pak Rano, 

Untuk eror yang terjadi berikut dikarenakan untuk pembuatan plotnya hanya membuat satu plot saja. Seperti dapat dilihat pada App.py baris ke 113, Pak Rano hanya membuat satu plot saja. sehingga proses render hanya dilakukan pada plot tersebut. 

Berikut saran perbaikan 

```
# First plot
ax1 = data_union['imdb_rating'].sort_values('imdb_rating',ascending=True).plot(kind='barh', figsize=(20,9))
figfile1 = BytesIO()
plt.savefig(figfile1, format='png', transparent=True)
figfile1.seek(0)
figdata_png1 = base64.b64encode(figfile1.getvalue())
plot_result1 = str(figdata_png1)[2:-1]
plt.close()


# Second plot
ax2 = data_union['metascore'].sort_values('metascore',ascending=True).plot(kind='barh', figsize=(20,9))
figfile2 = BytesIO()
plt.savefig(figfile2, format='png', transparent=True)
figfile2.seek(0)
figdata_png2 = base64.b64encode(figfile2.getvalue())
plot_result2 = str(figdata_png2)[2:-1]
plt.close()

# ... continue for other plots ...
```
Semoga jawaban ini membantu

Best Regards,
Team Algoritma
